Trying to compile on MacOS results in macro error

I'm seeing this when trying to do a fresh make on MacOS High Sierra. I don't know if it matters, but I installed my dependencies with brew instead of ports. Here is the specific error:


```
ccache g++ --std=c++14 -Werror -Wextra -Wall -Wfatal-errors -pedantic -Wno-overloaded-virtual -Wno-deprecated-register -march=native -fno-exceptions -fno-stack-protector -O3 -DNDEBUG -Iext/googletest/googletest/include -I. -I./ext/cl -c src/runtime/runtime.cc -o src/runtime/runtime.o
In file included from src/runtime/runtime.cc:46:
In file included from ./src/verilog/parse/parser.h:40:
In file included from ./src/verilog/parse/lexer.h:38:
verilog.tab.hh:1001:82: fatal error: too many arguments provided to function-like macro invocation
      basic_symbol (typename Base::kind_type t, YY_RVREF (std::pair<Identifier*, Maybe<RangeExpression>*>) v...
                                                                                 ^
verilog.tab.hh:74:10: note: macro 'YY_RVREF' defined here
# define YY_RVREF(Type)    Type&&
         ^
1 error generated.
make: *** [src/runtime/runtime.o] Error 1
```

--- a/src/verilog/parse/verilog.yy
+++ b/src/verilog/parse/verilog.yy
@@ -17,6 +17,12 @@
 
 namespace cascade {
 
+/* Typedefs, since Bison >3.1.2 uses a macro, and std::pair cannot be used. */
+typedef std::pair<bool, std::string> SignedNumber;
+typedef std::pair<size_t, NetDeclaration::Type> NetList;
+typedef std::pair<Identifier*, Maybe<RangeExpression>*> ModuleIdentifier;
+typedef std::pair<size_t, std::string> IdList;
+
 class Parser;
 
 } // namespace cascade
@@ -152,10 +158,10 @@ bool is_null(const cascade::Expression* e) {
 
 /* Numbers */
 %token <std::string> UNSIGNED_NUM
-%token <std::pair<bool, std::string>> DECIMAL_VALUE
-%token <std::pair<bool, std::string>> BINARY_VALUE
-%token <std::pair<bool, std::string>> OCTAL_VALUE
-%token <std::pair<bool, std::string>> HEX_VALUE
+%token <SignedNumber> DECIMAL_VALUE
+%token <SignedNumber> BINARY_VALUE
+%token <SignedNumber> OCTAL_VALUE
+%token <SignedNumber> HEX_VALUE
 
 /* Operator Precedence */
 %right QMARK COLON
@@ -249,7 +255,7 @@ bool is_null(const cascade::Expression* e) {
 %type <ArgAssign*> ordered_parameter_assignment
 %type <ArgAssign*> named_parameter_assignment
 %type <ModuleInstantiation*> module_instance
-%type <std::pair<Identifier*, Maybe<RangeExpression>*>> name_of_module_instance
+%type <ModuleIdentifier> name_of_module_instance
 %type <Many<ArgAssign>*> list_of_port_connections
 %type <ArgAssign*> ordered_port_connection
 %type <ArgAssign*> named_port_connection
@@ -378,7 +384,7 @@ bool is_null(const cascade::Expression* e) {
 %type <Many<ModuleItem>*> module_or_generate_item_S
 %type <Many<ArgAssign>*> named_parameter_assignment_P
 %type <Many<ArgAssign>*> named_port_connection_P
-%type <std::pair<size_t, NetDeclaration::Type>> net_type_L
+%type <NetList> net_type_L
 %type <NetDeclaration::Type> net_type_Q
 %type <Many<ModuleItem>*> non_port_module_item_S
 %type <Many<ArgAssign>*> ordered_parameter_assignment_P
@@ -391,7 +397,7 @@ bool is_null(const cascade::Expression* e) {
 %type <Maybe<RangeExpression>*> range_Q
 %type <size_t> reg_L
 %type <bool> signed_Q
-%type <std::pair<size_t, std::string>> simple_id_L
+%type <IdList> simple_id_L
 %type <Many<Statement>*> statement_S
 
 /* Alternate Rules */

